paper2remarkable/providers/diva.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+
+"""Provider for DiVA - Digitala Vetenskapliga Arkivet
+
+Author: G.J.J. van den Burg, Johan Holmberg
+License: See LICENSE file
+Copyright: 2019, 2024, G.J.J. van den Burg, Johan Holmberg
+
+"""
+
+import os
+import re
+import urllib.parse
+
+import bs4
+
+from ..exceptions import URLResolutionError, FulltextMissingError
+from ..log import Logger
+from ..utils import get_page_with_retry
+from ._base import Provider
+from ._info import Informer
+
+logger = Logger()
+
+
+class DiVAInformer(Informer):
+    def get_year(self, soup):
+        year = soup.find("meta", {"name": "citation_publication_date"}).get("content")
+        if not year:
+            logger.warning(
+                "Couldn't determine year information, maybe provide the desired filename using '--filename'?"
+            )
+            return ""
+        return year
+
+
+class DiVA(Provider):
+    re_abs = "^https?://[a-z]+.diva-portal.org/smash/record.jsf"
+    re_pdf = "^https?://[a-z]+.diva-portal.org/smash/get/diva2:[0-9]+/FULLTEXT"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.informer = DiVAInformer()
+
+    def _get_doc_url(self, abs_url):
+        page = get_page_with_retry(abs_url)
+        soup = bs4.BeautifulSoup(page, "html.parser")
+
+        pdf_url = soup.find("meta", {"name": "citation_pdf_url"})
+        print(pdf_url)
```

Comment:
Would you mind removing this `print` statement?

Reply:
Certainly!